src/main/java/main/Duke.java

```diff
@@ -0,0 +1,189 @@
+package main;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import tasks.Task;
+import tasks.Todo;
+import tasks.Deadline;
+import tasks.Event;
+
+
+public class Duke {
+
+    // Function to mark or unmark tasks
+    public static void userMarkOrUnmark(String command, String line, ArrayList<Task> list, int index, Task t) {
+
+        // User enters Mark, proceed to check if index is valid. If valid, then mark the task number
+        if (command.equals("mark")) {
+            try {
+                index = Integer.parseInt(line.substring(5));
+            } catch (NumberFormatException e) {
+                System.out.println("Task number is not a valid number");
+                return;
+            }
+            t = list.get(index - 1);
```

Comment:
Do not reuse formal parameters as local variables inside the method. This will also be taught in future lectures.